src/SprykerEco/Service/CrefoPayApi/HashCalculator/MacHashCalculator.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Service\CrefoPayApi\HashCalculator;
+
+use SprykerEco\Service\CrefoPayApi\CrefoPayApiConfig;
+
+class MacHashCalculator implements MacHashCalculatorInterface
+{
+    protected const HASHING_ALGORITHM = 'sha1';
+    protected const CHARS_TO_REPLACE = [" ", "\t", "\s", "\r", "\n"];
+
+    /**
+     * @var \SprykerEco\Service\CrefoPayApi\CrefoPayApiConfig
+     */
+    protected $config;
+
+    /**
+     * @param \SprykerEco\Service\CrefoPayApi\CrefoPayApiConfig $config
+     */
+    public function __construct(CrefoPayApiConfig $config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * @param array $data
+     *
+     * @return string
+     */
+    public function calculateMac(array $data): string
+    {
+        $macString = $this->createMacString($data);
+
+        return $this->hashValue($macString);
+    }
+
+    /**
+     * @param array $responseData
+     * @param string $mac
+     *
+     * @return bool
+     */
+    public function validateMac(array $responseData, string $mac): bool
+    {
+        return $this->calculateMac($responseData) === $mac;
+    }
+
+    /**
+     * @param array $data
```

Comment:
In general, one could be more specific in these cases: string[]

Reply:
It contains not only strings.